src/build.d

```diff
@@ -0,0 +1,1137 @@
+        auto hostDMDBase = hostDMDVer~"."~os;
+        auto hostDMDURL = "http://downloads.dlang.org/releases/2.x/"~hostDMDVer~"/dmd."~hostDMDBase;
+        env["HOST_DMD"] = hostDMDRoot.buildPath("dmd2", os, os == "osx" ? "bin" : "bin"~model, "dmd");
+        env["HOST_DMD_PATH"] = env["HOST_DMD"];
+        // TODO: use dmd.conf from the host too (in case there's a global or user-level dmd.conf)
+        env["HOST_DMD_RUN"] = env["HOST_DMD"];
+        if (!env["HOST_DMD"].exists)
+        {
+            writefln("Downloading DMD %s", hostDMDVer);
+            auto curlFlags = "-fsSL --retry 5 --retry-max-time 120 --connect-timeout 5 --speed-time 30 --speed-limit 1024";
+            hostDMDRoot.mkdirRecurse;
+            ("curl " ~ curlFlags ~ " " ~ hostDMDURL~".tar.xz | tar -C "~hostDMDRoot~" -Jxf - || rm -rf "~hostDMDRoot).spawnShell.wait;
+        }
+    }
+    else
+    {
+        env["HOST_DMD_PATH"] = ["which", env["HOST_DMD"]].execute.output.strip;
+        env["HOST_DMD_RUN"] = env["HOST_DMD"];
+    }
+
+    if (!env["HOST_DMD_PATH"].exists)
+    {
+        stderr.writefln("No DMD compiler is installed. Try AUTO_BOOTSTRAP=1 or manually set the D host compiler with HOST_DMD");
+        exit(1);
+    }
+
+    auto hostDMDVersion = [env["HOST_DMD_RUN"], "--version"].execute.output;
+    if (hostDMDVersion.find("DMD"))
+        env["HOST_DMD_KIND"] = "dmd";
+    else if (hostDMDVersion.find("LDC"))
+        env["HOST_DMD_KIND"] = "ldc";
+    else if (!hostDMDVersion.find("GDC", "gdmd")[0].empty)
+        env["HOST_DMD_KIND"] = "gdc";
+    else
+        enforce(0, "Invalid Host DMD found: " ~ hostDMDVersion);
+
+    env.getDefault("ENABLE_WARNINGS", "0");
+    string[] warnings;
+    if (env["ENABLE_WARNINGS"] != "0")
+    {
+        warnings = ["-Wall", "-Wextra", "-Werror",
+            "-Wno-attributes",
+            "-Wno-char-subscripts",
+            "-Wno-deprecated",
+            "-Wno-empty-body",
+            "-Wno-format",
+            "-Wno-missing-braces",
+            "-Wno-missing-field-initializers",
+            "-Wno-overloaded-virtual",
+            "-Wno-parentheses",
+            "-Wno-reorder",
+            "-Wno-return-type",
+            "-Wno-sign-compare",
+            "-Wno-strict-aliasing",
+            "-Wno-switch",
+            "-Wno-type-limits",
+            "-Wno-unknown-pragmas",
+            "-Wno-unused-function",
+            "-Wno-unused-label",
+            "-Wno-unused-parameter",
+            "-Wno-unused-value",
+            "-Wno-unused-variable",
+        ];
+        if (env["CXX_KIND"] == "g++")
+            warnings ~= [
+                "-Wno-logical-op",
+                "-Wno-narrowing",
+                "-Wno-unused-but-set-variable",
+                "-Wno-uninitialized",
+                "-Wno-class-memaccess",
+                "-Wno-implicit-fallthrough",
+            ];
+        else if (env["CXX_KIND"] == "clang++")
+            warnings ~= "-Wno-logical-op-parentheses";
```

Comment:
The check for `clang++` should be moved to the else branch of `env["ENABLE_WARNINGS"] != "0"`.